service/service.go

```diff
@@ -460,3 +471,57 @@ func parseClusterIPRange(ipRange string) (net.IP, net.IP, error) {
 
 	return networkIP, apiServerIP, nil
 }
+
+func migrateSecretLabels(logger micrologger.Logger, k8sClient kubernetes.Interface) error {
+	var secrets []*corev1.Secret
+	{
+		o := metav1.ListOptions{
+			LabelSelector: "clusterKey=encryption",
+		}
+
+		l, err := k8sClient.CoreV1().Secrets(corev1.NamespaceAll).List(o)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		for _, s := range l.Items {
+			secrets = append(secrets, s.DeepCopy())
+		}
+	}
+
+	for _, s := range secrets {
+		if hasLabels(s, label.Cluster, label.RandomKey) {
+			continue
+		}
+
+		s.Labels[label.Cluster] = s.Labels["clusterID"]
+		s.Labels[label.RandomKey] = label.RandomKeyTypeEncryption
+
+		_, err := k8sClient.CoreV1().Secrets(s.Namespace).Update(s)
```

Comment:
Should be fine here as these secrets aren't changed too often. In general I would prefer a JSON Patch request with validation of the resource version.

Reply:
The resource version is in the secret that we just fetched. The update fails when the version changed meanwhile. I do not care about the patch in this case to be honest.